File: problem-sets/problem-set-01/cash/cash.c
#include <stdio.h>

int main(void) {
  int change = -1;
  do {
    printf("\nChange owed (in cents): ");
    scanf("%d", &change);

    if (change < 0) {
      printf("Please enter a non-negative amount.\n");
      continue;
    }

    //How many quarters are needed
    int quarters = change / 25;
    //Update change to reflect remaining cents
    change = change % 25;

    //How many dimes are needed
    int dimes = change / 10;
    change = change % 10;

    //How many nickels are needed
    int nickels = change / 5;
    change = change % 5;

    //Remaining cents are pennies
    int pennies = change;

    //Calculate total coins
    int totalCoins = quarters + dimes + nickels + pennies;

    printf("Total coins needed: %d\n", totalCoins);

  } while (change != 0);

  return 0;
}

File: problem-sets/problem-set-06/sentimental-cash/cash.py
while True:
  try:
      dollar = float(input("Change (in dollars): "))
      if dollar >= 0:
          break
      else:
          print("Please enter a positive number")
  except ValueError:
      print("Please enter a valid number")

cents = round(dollar * 100)

quarters = cents//25
## Update change to reflect remaining amount
cents = cents % 25

dimes = cents//10
cents = cents % 10

nickels = cents//5    
cents = cents % 5

## Remaining change is all in pennies
pennies = cents

total_coins = quarters + dimes + nickels + pennies
print(total_coins)
